e2e/_suites/kubernetes-autodiscover/features/beats.feature

```diff
@@ -0,0 +1,38 @@
+@beats
+Feature: Beats
+  Use Kubernetes autodiscover features in Beats to monitor pods
+
+Scenario: Pod is started
+  Given "filebeat" is running with "hints enabled"
+   When "a pod" is deployed
+   Then "filebeat" collects events with "kubernetes.pod.name:a-pod"
+
+Scenario: Pod is deleted
+  Given "filebeat" is running with "hints enabled"
+    And "a pod" is deployed
+    And "filebeat" collects events with "kubernetes.pod.name:a-pod"
```

Comment:
Does the order of these steps change the expected behaviour?

Reply:
The first one for what happens to filebeat if it starts collecting events before a pod is deployed, and the second one, exactly as you proposed.

Reply:
Yes, order matters here, I want to be sure that the beat was collecting events before the pod was deleted, and that after deleting it, the beat stops collecting events.

(While writing this I see this scenario doesn't make any sense for filebeat, a deleted pod is not going to generate logs that could be collected by filebeat. But it makes sense for metricbeat that will generate events with errors if it tries to continue monitoring a deleted pod).